kinoml/ml/torch_models.py

```diff
@@ -179,28 +183,182 @@ def __init__(
     ):
         super().__init__()
 
-        self.nb_char = nb_char
-        self.max_length = max_length
+        self.input_shape = input_shape
         self.embedding_shape = embedding_shape
         self.kernel_shape = kernel_shape
         self.hidden_shape = hidden_shape
         self.output_shape = output_shape
         self._activation = activation
 
         self.convolution = nn.Conv1d(
-            in_channels=self.nb_char,
+            in_channels=self.input_shape[0], # nb of characters
             out_channels=self.embedding_shape,
             kernel_size=self.kernel_shape,
         )
-        self.temp = (self.max_length - self.kernel_shape + 1) * self.embedding_shape
+        self.temp = (self.input_shape[1] - self.kernel_shape + 1) * self.embedding_shape
         self.fully_connected_1 = nn.Linear(self.temp, self.hidden_shape)
         self.fully_connected_out = nn.Linear(self.hidden_shape, self.output_shape)
 
     def forward(self, x):
         """
         Defines the foward pass for a given input 'x'
         """
+        x = x[0]
```

Comment:
Can we document what shape is expected for `x` as input, and why this line (`x = x[0]`) is appropriate here?

Reply:
The reason why we access the first (or 2 first, or more) elements of `x` is due to the way the parquet files are constructed.
I added details and hope it makes it clearer.

Here's the documentation link: 
https://github.com/openkinome/experiments-binding-affinity/blob/master/features/featurize-template.ipynb , 
and paragraph:



Write tensors to disk
Output files are written to _output/<PIPELINE>/<DATASET>/<GROUP>.parquet files.
Each parquet will contain at least two array-like objects. The dimensionality of the parquet files is built as (systems, X_or_y, ...). For example, the first X vector for the first system is accessed like parquet[0, "0"]. Notice how the 2nd index is a string! (awkward design).

"0" (X, featurized systems). See DatasetProvider.to_awkward for more info.
"1" (y, associated measurements)
If X is composed of more than one array (e.g. connectivity matrix + node features), these are flattened to "0", "1", "2", and so on. y is ALWAYS the last one in that list (accessible via data.fields)